### PR TITLE
Remove OpenVX

### PR DESCRIPTION
**Pull request by invarrow**
resolves https://github.com/opencv/opencv/issues/24995

OpenCV cleanup https://github.com/opencv/opencv/issues/25007

### CODE REVIEW

File: 3rdparty/carotene/include/carotene/functions.hpp [hunk 1]
```diff
@@ -420,7 +420,7 @@ namespace CAROTENE_NS {
 
     /*
         For each point `p` within `size`, do:
-        dst[p] = scale / src[p] 
+        dst[p] = scale / src[p]
```
Comment (dkurt) [inline 3rdparty/carotene/include/carotene/functions.hpp:423]: unrelated file. Please revert if possible

File: platforms/js/build_js.py [hunk 2]
```diff
@@ -103,7 +103,7 @@ def get_cmake_cmd(self):
                "-DWITH_WEBP=OFF",
                "-DWITH_OPENEXR=OFF",
                "-DWITH_OPENGL=OFF",
-               "-DWITH_OPENVX=OFF",
+#               "-DWITH_OPENVX=OFF",
```
Comment (dkurt) [inline platforms/js/build_js.py:106]: Feel free to remove, not comment

File: cmake/templates/xcode-launch-c.in [hunk 3]
```diff
@@ -7,5 +7,5 @@ if [[ "$1" = "${CMAKE_C_COMPILER}" ]] ; then
     shift
 fi
 
-export CCACHE_CPP2=true
+export CCACHE_CPP2=false
```
Comment (dkurt) [inline cmake/templates/xcode-launch-c.in:10]: That won't make things better. Try add `timeout-minutes` to https://github.com/opencv/opencv/blob/5.x/.github/workflows/PR-5.x.yaml:

```yaml
  macOS-x64:
    uses: opencv/ci-gha-workflow/.github/workflows/OCV-PR-5.x-macOS-x86_64.yaml@main
    timeout-minutes: 90
```

File: .github/workflows/PR-5.x.yaml [hunk 4]
```diff
@@ -38,6 +38,7 @@ jobs:
 
   macOS-x64:
     uses: opencv/ci-gha-workflow/.github/workflows/OCV-PR-5.x-macOS-x86_64.yaml@main
+    timeout-minutes: 90
```
Comment (dkurt) [inline .github/workflows/PR-5.x.yaml:41]: @invarrow, sorry, that won't work. Please remove that line, I changed timeout here: https://github.com/opencv/ci-gha-workflow/pull/164.

File: modules/core/include/opencv2/core/utils/trace.hpp [hunk 5]
```diff
@@ -106,7 +106,7 @@ enum RegionLocationFlag {
 
     REGION_FLAG_IMPL_IPP = (1 << 16),            ///< region is part of IPP code path
     REGION_FLAG_IMPL_OPENCL = (2 << 16),         ///< region is part of OpenCL code path
-    REGION_FLAG_IMPL_OPENVX = (3 << 16),         ///< region is part of OpenVX code path
+///    REGION_FLAG_IMPL_OPENVX = (3 << 16),         ///< region is part of OpenVX code path
```
Comment (dkurt) [inline modules/core/include/opencv2/core/utils/trace.hpp:109]: @mshabunin, @opencv-alalek, shall we keep this value for compatibility?

Comment (mshabunin) [reply modules/core/include/opencv2/core/utils/trace.hpp:109]: I think we can remove it. Since it is 5.x we can break compatibility.

Comment (dkurt) [reply modules/core/include/opencv2/core/utils/trace.hpp:109]: @invarrow, please remove the line completely

Comment (invarrow) [reply modules/core/include/opencv2/core/utils/trace.hpp]: done

**Review by dkurt — APPROVED** [review]

File: samples/CMakeLists.txt [hunk 6]
```diff
@@ -33,9 +33,6 @@ endif()
 if((NOT ANDROID) AND HAVE_OPENGL)
   add_subdirectory(opengl)
 endif()
-if(HAVE_OPENVX)
-  add_subdirectory(openvx)
```
Comment (opencv-alalek) [inline samples/CMakeLists.txt:37]: Sanity check still shows OpenVX reference on the line 131:

```
$ grep -Rni openvx ./
```

Comment (invarrow) [reply samples/CMakeLists.txt:37]: addressed👍🏽

**Comment by mshabunin** [comment]
I suggest postponing merge of this PR until after regular 4.x->5.x merge.

**Review by asmorkalov — APPROVED** [review]
:+1:
